app/controllers/posts_controller.rb

```diff
@@ -22,14 +23,17 @@ def create
 
   def update
     if post.update!(post_params)
+      BroadcastService.broadcast(post: post, user: current_user, action: :updated)
       render json: { status: 200, post: @post }
     else
       render json: { status: 500, errors: @post.errors.full_messages }
     end
   end
 
   def destroy
+    deleted_post = post
     if post.destroy
+      BroadcastService.broadcast(post: deleted_post, user: current_user, action: :destroyed)
```

Comment:
You can avoid doing `deleted_post = post`
and use post directly as the 1st parameter of the broadcast service

Reply:
The post is destroyed, so I cannot retrieve information of it. Check my private method.